pmd-core/src/main/java/net/sourceforge/pmd/lang/rule/ParametricRuleViolation.java

```diff
@@ -183,11 +192,31 @@ public String getVariableName() {
         return variableName;
     }
 
+    @Experimental
+    @Override
+    public String getSourceCode(String encoding) {
+        try {
+            SourceCode sourceCode = new SourceCode(new SourceCode.StringCodeLoader(rawCodeSnippet));
+            return sourceCode.getSlice(getBeginLine(), getEndLine());
+
+        } catch (RuntimeException e) {
+            return null;
```

Comment:
I also wonder how much of a feature that is, if it's so fragile. Why would your code here fail?

Reply:
I had added this when I was using SourceCode.FileCodeLoader as it was causing an exception.